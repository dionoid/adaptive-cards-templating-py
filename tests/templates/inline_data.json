{
    "type": "AdaptiveCard",
    "$data": {
        "employee": {
            "name": "Matt",
            "manager": { "name": "Thomas" },
            "peers": [{
                "name": "Andrew" 
            }, { 
                "name": "Lei"
            }, { 
                "name": "Mary Anne"
            }, { 
                "name": "Adam"
            }]
        }
    },
    "body": [
        {
            "type": "TextBlock",
            "text": "Hi ${employee.name}! Here's a bit about your org..."
        },
        {
            "type": "TextBlock",
            "text": "Your manager is: ${employee.manager.name}"
        },
        {
            "type": "TextBlock",
            "text": "The first 3 of your peers are: ${employee.peers[0].name}, ${employee.peers[1].name}, ${employee.peers[2].name}"
        },
        {
            "type": "Container",
            "$when": "${if(employee.peers.length > 3, true, false)}",
            "items": [
                {
                    "type": "TextBlock",
                    "text": "You have more than 3 peers!"
                },
                {
                    "type": "FactSet",
                    "facts": [
                        {
                            "$data": "${employee.peers}",
                            "title": "${name}"
                        }
                    ]
                }
            ]
        }
    ]
}