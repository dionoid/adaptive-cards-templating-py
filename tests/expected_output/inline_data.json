{
  "type": "AdaptiveCard",
  "body": [
    {
      "type": "TextBlock",
      "text": "Hi Matt! Here's a bit about your org..."
    },
    {
      "type": "TextBlock",
      "text": "Your manager is: Thomas"
    },
    {
      "type": "TextBlock",
      "text": "The first 3 of your peers are: Andrew, Lei, Mary Anne"
    },
    {
      "type": "Container",
      "items": [
        {
          "type": "TextBlock",
          "text": "You have more than 3 peers!"
        },
        {
          "type": "FactSet",
          "facts": [
            {
              "title": "Andrew"
            },
            {
              "title": "Lei"
            },
            {
              "title": "Mary Anne"
            },
            {
              "title": "Adam"
            }
          ]
        }
      ]
    }
  ]
}